### PR TITLE
[FIX] Add margin and check to AssetActionButton text

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

The `AssetActionButton` component wasn't handling long string. This PR add a margin and a 10 character max check to the component to avoid UI issues

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Screenshots/Recordings**

| BUG | FIX |
|-----|-----|
| Bug | Fix |

**Issue**

Progresses [Issue Reference](https://github.com/MetaMask/mobile-planning/issues/278#issuecomment-1131975445)

### CODE REVIEW

File: app/components/UI/AssetActionButton/index.js [hunk 1]
```diff
@@ -123,7 +126,9 @@ function AssetActionButton({ onPress, icon, label, disabled }) {
         {icon && (typeof icon === 'string' ? getIcon(icon) : icon)}
       </View>
       <Text centered style={styles.buttonText} numberOfLines={1}>
-        {label}
+        {label.size > maxStringLength
+          ? `${label[(0, maxStringLength - 3)]}...`
```
Comment (Fatxx) [inline app/components/UI/AssetActionButton/index.js:130]: Didn't you want to `.substring()` the value instead?

**Review by Fatxx — COMMENTED** [review]
Shouldn't this be pointing to `release/5.2`?

**Comment by gantunesr** [comment]
@Fatxx I need this in main for another development, I'll cherry pick it to the release branch

File: app/components/UI/AssetActionButton/index.js (hunk 1 above)
Comment (gantunesr) [reply app/components/UI/AssetActionButton/index.js]: Updated in [4b934bd](https://github.com/MetaMask/metamask-mobile/pull/4381/commits/4b934bdcbdda600f7cc06bb2c5d7b0619d95cf4c)

**Review by Fatxx — COMMENTED** [review]
LGTM

**Review by Fatxx — APPROVED** [review]

File: app/components/UI/AssetActionButton/__snapshots__/index.test.tsx.snap [hunk 2]
```diff
@@ -285,12 +294,15 @@ exports[`AssetActionButtons should render type receive correctly 1`] = `
...
         "fontSize": 14,
+        "marginHorizontal": 3,
         "marginTop": 8,
       }
     }
     underline={false}
     upper={false}
-  />
+  >
+    receive receive
```
Comment (Fatxx) [inline app/components/UI/AssetActionButton/__snapshots__/index.test.tsx.snap:304]: Didn't seem to work, uhm?

File: app/components/UI/AssetActionButton/index.js [hunk 3]
```diff
@@ -123,7 +126,9 @@ function AssetActionButton({ onPress, icon, label, disabled }) {
         {icon && (typeof icon === 'string' ? getIcon(icon) : icon)}
       </View>
       <Text centered style={styles.buttonText} numberOfLines={1}>
-        {label}
+        {label.size > maxStringLength
```
Comment (Fatxx) [inline app/components/UI/AssetActionButton/index.js:129]: It's `label.length` instead I suppose

**Review by Fatxx — CHANGES_REQUESTED** [review]

File: app/components/UI/AssetActionButton/__snapshots__/index.test.tsx.snap (hunk 2 above)
Comment (gantunesr) [reply app/components/UI/AssetActionButton/__snapshots__/index.test.tsx.snap]: [95f3990](https://github.com/MetaMask/metamask-mobile/pull/4381/commits/95f399064be7804d9a725c97f265079659213763)

**Review by Fatxx — APPROVED** [review]

**Review by chrisleewilcox — APPROVED** [review]
LGTM!

**Pull request merged**